sklearn/linear_model/_base.py

```diff
@@ -350,7 +350,7 @@ def predict(self, X):
         xp, _ = get_namespace(X)
         scores = self.decision_function(X)
         if len(scores.shape) == 1:
-            indices = xp.astype(scores > 0, int)
+            indices = xp.astype(scores > 0, xp.int64)
```

Comment:
```suggestion
            # Casting to int32 is necessary to make this code work on
            # 32 bit platforms such as pyodide but limits the number of
            # samples we can classify to a few billions at a time on 64
            # bit architectures. In 2024 it does not feel like users can
            # reach this limit. If that ever happens this line should be
            # updated to make it platform dependent instead.
            # For now, we refrain to do so for the sake of keeping the
            # code simple.
            indices = xp.astype(scores > 0, xp.int32)
```

Reply:
👍 The best workaround for the time being.

Reply:
Actually, https://github.com/scikit-learn/scikit-learn/pull/28555#discussion_r1522978780 might be better. WDYT?

Reply:
Implemented the alternative in 46a3936.